server/appengine/src/main/java/who/Environment.java

```diff
@@ -51,10 +51,13 @@ public static Environment current() {
 
   private static String getApplicationId() {
     String applicationId = AppEngine.applicationId();
-    // "o~" prefix bug in App Engine. Not sure where those 2 chars at the beginning come from.
-    final String PREFIX_BUG = "o~";
-    if (applicationId.startsWith(PREFIX_BUG)) {
-      applicationId = applicationId.substring(PREFIX_BUG.length());
+    if (applicationId == null) {
+      return null;
+    }
+    // Strip shard.
+    int tilde = applicationId.indexOf('~');
+    if (tilde >= 0) {
+      applicationId = applicationId.substring(tilde + 1);
     }
     return applicationId;
```

Comment:
Extract to method, so it can be tested without needing injection.
```
assertEquals(stripShard(null), null);
assertEquals(stripShard("o~AppEngineID"), "AppEngineID");
assertEquals(stripShard("dev~AppEngineID"), "AppEngineID");
```

Reply:
I don't want to change this part of the code beyond the minimum right now. This is really a bug here:
https://github.com/presentco/present-engine/blob/master/src/main/java/present/engine/AppEngine.java#L36